File: scratch/inspect_secom_sqlite.js
const sqlite3 = require('sqlite3').verbose();
const path = require('path');

const dbPath = "C:\\(주)에스원\\세콤매니저(근태·식당) v9.0.1\\DB\\WorkManager.DB";

const db = new sqlite3.Database(dbPath, sqlite3.OPEN_READONLY, (err) => {
  if (err) {
    console.error("Failed to open SQLite database:", err.message);
    process.exit(1);
  }
  console.log("Successfully connected to WorkManager.DB!");
});

db.serialize(() => {
  // Get all table names
  db.all("SELECT name FROM sqlite_master WHERE type='table'", [], (err, rows) => {
    if (err) {
      console.error(err.message);
      return;
    }
    console.log("\n=== Tables in WorkManager.DB ===");
    console.log(rows.map(r => r.name).join(', '));
    
    // Look for tables containing 'ERP', 'Link', 'Query', 'Schedule', 'Time', 'Send'
    const interestingTables = rows
      .map(r => r.name)
      .filter(name => /erp|link|query|schedule|time|send/i.test(name));
      
    console.log("\n=== Interesting Tables ===");
    console.log(interestingTables);
    
    // If there is an ERP or SecomLink related table, dump its schema and content
    if (rows.some(r => r.name === 't_SecomLinkQuery')) {
      queryTable('t_SecomLinkQuery');
    }
    if (rows.some(r => r.name === 't_SecomLinkOption')) {
      queryTable('t_SecomLinkOption');
    }
    if (rows.some(r => r.name === 't_ErpSetting')) {
      queryTable('t_ErpSetting');
    }
    if (rows.some(r => r.name === 't_Scheduler')) {
      queryTable('t_Scheduler');
    }
    
    // Fallback: search for schemas of tables that look interesting
    interestingTables.forEach(tbl => {
      queryTable(tbl);
    });
  });
});

function queryTable(tableName) {
  db.all(`PRAGMA table_info(${tableName})`, [], (err, columns) => {
    if (err) {
      console.error(`Error reading schema for ${tableName}:`, err.message);
      return;
    }
    console.log(`\n--- Schema of ${tableName} ---`);
    console.table(columns.map(c => ({ Column: c.name, Type: c.type })));
    
    db.all(`SELECT * FROM ${tableName} LIMIT 10`, [], (err, rows) => {
      if (err) {
        console.error(`Error querying ${tableName}:`, err.message);
        return;
      }
      console.log(`--- Content of ${tableName} (max 10 rows) ---`);
      console.log(rows);
    });
  });
}

File: scratch/inspect_secom_sqlite.py
import sqlite3
import os

db_path = "C:\\(주)에스원\\세콤매니저(근태·식당) v9.0.1\\DB\\WorkManager.DB"

if not os.path.exists(db_path):
    print(f"Error: Database file does not exist at: {db_path}")
    exit(1)

conn = sqlite3.connect(db_path)
cursor = conn.cursor()

# Get all table names
cursor.execute("SELECT name FROM sqlite_master WHERE type='table'")
tables = [row[0] for row in cursor.fetchall()]

print("\n=== Tables in WorkManager.DB ===")
print(", ".join(tables))

interesting_keywords = ['erp', 'link', 'query', 'schedule', 'time', 'send', 'option', 'setting']
interesting_tables = [t for t in tables if any(k in t.lower() for k in interesting_keywords)]

print("\n=== Interesting Tables ===")
print(interesting_tables)

def query_table(table_name):
    print(f"\n--- Schema of {table_name} ---")
    cursor.execute(f"PRAGMA table_info({table_name})")
    cols = cursor.fetchall()
    for col in cols:
        print(f"  {col[1]} ({col[2]})")
        
    print(f"--- Content of {table_name} (max 20 rows) ---")
    cursor.execute(f"SELECT * FROM {table_name} LIMIT 20")
    rows = cursor.fetchall()
    for r in rows:
        print(r)

for tbl in interesting_tables:
    query_table(tbl)

conn.close()
